### PR TITLE
new PR for the PostgreSQL Direct Queries

### PR DESCRIPTION
**Pull request by jaskeerat789**

### CODE REVIEW

**Comment by jaskeerat789** [comment]
@wtrocki I hope now things are at there place

**Comment by wtrocki** [comment]
Unit tests are missing. Additionally we will need helper to consume this class as the other data providers. Finally we should move to use this provider in sample app

**Comment by jaskeerat789** [comment]
Thanks for update sir

**Comment by jaskeerat789** [comment]
I'll return to you asap

File: packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    this.tableMap = buildModelTableMap(baseType);
+    this.tableName = this.tableMap.tableName;
+  }
+
+  public async create(data: Type): Promise<Type> {
+    const { data: createData } = getDatabaseArguments(this.tableMap, data);
+
+    //tslint:disable-next-line: await-promise
+    // const [id] = await this.db(this.tableName).insert(createData);
+    const [id] = await this.db.query(`INSERT INTO ${this.tableName} VALUES (${createData})`);
```
Comment (craicoverflow) [inline packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts]: `createData` is a key-value object, have you tested this?

File: packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ * - Sqlite
+ * - MySQL (MariaDB)
+ * - Postgress (by `PgKnexDBDataProvider`)
+ *
+ * NOTE: For Postgres use dedicated `PgKnexDBDataProvider` that implements more performant creation method.
+ */
+//tslint:disable-next-line: no-any
+export class PostgreDBDataProvider<Type = any, GraphbackContext = any> implements GraphbackDataProvider<Type, GraphbackContext>{
+
+  protected db: Db;
```
Comment (craicoverflow) [inline packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts]: What is this object? It is not imported anywhere.

File: packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    //tslint:disable-next-line: await-promise
+    // const dbResult = await this.db.select().from(this.tableName).where(this.tableMap.idField, '=', id)
+    const dbResult = await this.db.query(`SELECT * FROM ${this.tableName} WHERE ${this.tableMap.idField} =  ${id}`);
+    if (dbResult && dbResult[0]) {
+      return dbResult[0]
+    }
+    throw new NoDataError(`Cannot create ${this.tableName}`);
+  }
+
+  public async update(data: Type): Promise<Type> {
```
Comment (craicoverflow) [inline packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts]: `updateData` is not used anywhere, this does two queries.

File: packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
+import { buildModelTableMap, getDatabaseArguments, ModelTableMap } from '@graphback/core';
+import { GraphQLObjectType } from 'graphql';
+// import * as Knex from 'knex';
+import { GraphbackDataProvider, GraphbackPage, NoDataError, AdvancedFilter } from '@graphback/runtime';
+
+/**
+ * Knex.js database data provider exposing basic CRUD operations that works with all databases that knex supports.
```
[the remark this answers is not in the document]
Comment (jaskeerat789) [reply packages/graphback-runtime-knex/src/PostgreSQLDataProvider.ts]: Ok, @craicoverflow ... I was thinking of creating an entirely new package

**Comment by wtrocki** [comment]
@jaskeerat789 Sorry for closing this PR. We have some internal chat and decided that we will hold of with PostgreSQL integration until we make some other internal changes. Thank you so much for contribution

**Pull request closed without merging**